tests/integration/test_interface.py

```diff
@@ -140,26 +142,35 @@ def test_generate_dataset_and_col_noising(
         assert check_noised.loc[originally_missing_idx, col_name].isna().all()
 
         # Check for noising where applicable
-        shared_not_missing_idx = shared_idx.difference(originally_missing_idx)
+        to_compare_idx = shared_idx.difference(originally_missing_idx)
         if col.noise_types:
             assert (
-                check_original.loc[shared_not_missing_idx, col_name].values
-                != check_noised.loc[shared_not_missing_idx, col_name].values
+                check_original.loc[to_compare_idx, col_name].values
+                != check_noised.loc[to_compare_idx, col_name].values
             ).any()
+
+            noise_level = (
+                check_original.loc[to_compare_idx, col_name].values
+                != check_noised.loc[to_compare_idx, col_name].values
+            ).mean()
+
             # Check that the amount of noising seems reasonable
-            threshold = 1 - cell_probability ** len(col.noise_types)
-            # NOTE: The threshold is not perfect - it is very underconservative
-            #   because it does not account for token-level noising which could
-            #   result in no actual changes to a value despite being chosen
-            # TODO: generate more accurate thresholds
-            assert (
-                check_original.loc[shared_not_missing_idx, col_name].values
-                != check_noised.loc[shared_not_missing_idx, col_name].values
-            ).mean() <= threshold
-        else:
+            tmp_config = config[dataset.name][Keys.COLUMN_NOISE][col_name]
+            includes_token_noising = [
+                c for c in tmp_config if Keys.TOKEN_PROBABILITY in tmp_config[c].keys()
+            ]
+            # NOTE: The threshold assigned when we have token-level noising
+            # is guessed at since it's difficult to calculate.
+            # TODO: Come up with a more accurate values. There are token probabilities
+            # and additional parameters to consider as well as the rtol when the
+            # number of compared is small.
+            expected_noise = 1 - (1 - cell_probability) ** len(col.noise_types)
+            rtol = 0.5 if includes_token_noising else 0.11
```

Comment:
They can be off by as much as 50%? I think it's fine, but hopefully Zeb's PR [here](https://github.com/ihmeuw/vivarium_census_prl_synth_pop/pull/252) bears fruit and lets us do better.